## 生物統計與流行病學 作業五
## Biostatistics and Epidemiology HW5
B10401019 呂祐寬


### 第一大題(共40分)：每小題10分。
### First Part (Total 40 points): 10 points each


令 X 表示個體之抽菸狀態，即個體有抽菸則X=1，沒抽菸則X=0。G=1表示第1組，G=2表示第2組。這兩組共收集到20筆的抽菸狀態的個別資料，如下所列: <br>
Let X represents for the smokers status of an individual, that is X=1 if the person smokes, and X=0 if the person does not. G=1 and G=2 stand for group 1 and group 2, respectively. The table below shows total 20 observations of the smokers status of individuals from two groups:

| id | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 | 1 | 2 | 3 | 4 | 5 | 6 | 7 | 8 | 9 | 10 |
| -- | - | - | - | - | - | - | - | - | - | -- | - | - | - | - | - | - | - | - | - | -- |
| G  | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1 | 1  | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2 | 2  |
| X  | 1 | 1 | 0 | 1 | 0 | 1 | 1 | 1 | 1 | 1  | 0 | 1 | 1 | 1 | 0 | 0 | 0 | 1 | 0 | 0  |


<font color="#fcba03">

Q1. 若G=1,2是兩組獨立樣本，G=1代表有得怪病Z、G=2代表沒有得怪病Z，即相同id但不同組別代表不同個體，則X則是觀察每組每個個體之抽菸狀態，請將資料整理成一個2x2 列聯表。 <br>
Q1. If G=1 and G=2 are independent samples, G=1 stands for having disease Z and G=0 stands for not having disease Z, which means the same id from two different groups are not the same person. X is the smokers status of each individual. Please make a 2 by 2 contingency table using this dataset.


</font>

<font color="#039dfc">

### Ans
By the R code below

</font>

In [1]:
library(plyr) # chatGPT
G_raw <- c(1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 2, 2, 2, 2, 2)
G <- mapvalues(G_raw, c(1, 2), c("Z(+)", "Z(-)"))
X_raw <- c(1, 1, 0, 1, 0, 1, 1, 1, 1, 1, 0, 1, 1, 1, 0, 0, 0, 1, 0, 0)
X <- mapvalues(X_raw, c(0, 1), c("non-smokers", "smokers"))
df1 <- data.frame(G, X)

table1 <- table(df1$G, df1$X)
table1

      
       non-smokers smokers
  Z(-)           6       4
  Z(+)           2       8

<font color="#039dfc">

we can get the contingency table by calculating the sums of each row and column.

|      | smokers | non-smokers | Total |
| ---- | ------- | ----------- | ----- |
| Z(+) | $8$     | $2$         | $10$  |
| Z(-) | $4$     | $6$         | $10$  |
|Total | $12$    | $8$         | $20$  |  

</font>

<font color="#fcba03">

Q2.	根據列聯表估計抽菸得怪病Z的odds ratio。<br>
Q2. Estimate the odds ratio for having disease Z from the contingency table.

</font>

<font color="#039dfc">

### Ans
The odds ratio for smokers against having disease Z from the contingency table can be calculated using the contigency table above by:
 $$\mathrm{Odds\;ratio} = \frac{8 \times 6}{2 \times 4}=6$$

</font>

<font color="#fcba03">

Q3.	請根據抽菸對怪病Z是否存在相關性的統計問題，寫出合適的虛無和對立假說。 <br>
Q3. Please write down the appropriate null hypothesis and alternative hypothesis, according to the statistical problem on whether the correlation exist between smokers and disease Z.


</font>

<font color="#039dfc">

### Ans
$H_0$: There is no correlation between smokers and disease Z. (The observed correlation is due to chance) <br>
$H_1$: There is a correlation between smokers and disease Z. (The observed correlation is not due to chance)

</font>

<font color="#fcba03">

Q4.	請計算合適的檢定統計量並做出判斷。<br>
Q4. Please calculate the appropriate test statistics and make decision

</font>

<font color="#039dfc">

### Ans
Having the contingency table, we can conduct the $\chi^2$ test

</font>

In [2]:
chi_square1 <- chisq.test(table1)
chi_square1



Warning message in chisq.test(table1):
“Chi-squared approximation may be incorrect”



	Pearson's Chi-squared test with Yates' continuity correction

data:  table1
X-squared = 1.875, df = 1, p-value = 0.1709


<font color="#039dfc">

From the data, we get the p-value is $0.1709$ , which is larger than $0.05$. Thus, we are unable to reject the null hypothesis.


However, upon getting the warning:
```
Warning message in chisq.test(table1):
“Chi-squared approximation may be incorrect”
```
we seem to have a very small data, which results in the unsure detection of the null hypothesis.

Therefore, since we are using R, I will also do the Fisher's exact test as well.

</font>


In [3]:
fisher_test1 <- fisher.test(table1)
fisher_test1


	Fisher's Exact Test for Count Data

data:  table1
p-value = 0.1698
alternative hypothesis: true odds ratio is not equal to 1
95 percent confidence interval:
  0.6026805 79.8309210
sample estimates:
odds ratio 
  5.430473 


<font color="#039dfc">

The p-value is $0.1698$, which is also larger than $0.05$. Thus, the result is same as chi_square test, which indicates that we are not able to reject the null hypothesis.

</font>

### 第二大題(共30分)：每小題10分。
### Second Part (Total 30 points): 10 points each

一小型臨床研究欲測試新療法是否能有效增加病患對藥物之敏感度。受限於研究資源不多，僅能收集12名受試者進行測試。該研究為了克服樣本數的限制，採取crossover的設計，使12名受試者皆能測得新療法與傳統療法的藥物敏感度，並精巧的設計了適當的washout period降低藥物交互作用與干擾作用。其測試之反應程度如下表，請問新療法與傳統療法敏感度是否有顯著差異? (α=0.05) <br>
A small-scale clinical research wanted to examine whether the new treatment could increase the patient’s sensitivities on drugs effectively or not. Limited to the research resources, it only collected 12 participants. To overcome the limitation of sample size, it adopted the crossover design, and all 12 participants could measure the sensitivities on both new treatment and tradition treatment. They delicately designed an appropriate washout period to reduce the interaction and confounding effect. The reactions of the examinations are listed in the table below. Is there significant difference in sensitivities between new treatment and the traditional treatment? (α=0.05)

| ID                 | 1  | 2  | 3  | 4  | 5  | 6  | 7  | 8  | 9  | 10 | 11 | 12 |
| ------------------ | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- | -- |
| 新療法 (new)          | 74 | 61 | 73 | 66 | 48 | 45 | 75 | 57 | 71 | 72 | 69 | 80 |
| 傳統療法 (traditional) | 64 | 79 | 67 | 62 | 88 | 67 | 82 | 43 | 70 | 68 | 77 | 63 |


<font color="#fcba03">

Q5.	請根據提議設定虛無假設與對立假設 <br>
Q5. Please state the null hypothesis and alternative hypothesis according to the question.

</font>

<font color="#039dfc">

$H_0$: There is no difference in medium of sensitivities between the new treatment and the traditional treatment. <br>
$H_1$: There is difference in medium of sensitivities between the new treatment and the traditional treatment.

</font>

<font color="#fcba03">

Q6.	請判斷此題合適的統計方法為何者？ <br>
Q6. Please decide an appropriate statistical method for this problem.

</font>

<font color="#039dfc">

### Ans
With a small dataset and no appearant outliers in the dataset (by eyeballing), I decide to use the **Wilcoxon Signed-Rank test** for its larger statistical power.



</font>

<font color="#fcba03">

Q7.	計算檢定統計量並做出判斷 <br>
Please calculate the test statistic and make judgement.


</font>

<font color="#039dfc">

### Ans


</font>

In [4]:
library(stats)

new_treatment <- c(74, 61, 73, 66, 48, 45, 75, 57, 71, 72, 69, 80)
traditional_treatment <- c(64, 79, 67, 62, 88, 67, 82, 43, 70, 68, 77, 63)

wilcox_test2 <- wilcox.test(new_treatment, traditional_treatment, paired = TRUE, alternative = "two.sided", conf.level = 0.95)

print(wilcox_test2)


Warning message in wilcox.test.default(new_treatment, traditional_treatment, paired = TRUE, :
“cannot compute exact p-value with ties”



	Wilcoxon signed rank test with continuity correction

data:  new_treatment and traditional_treatment
V = 34, p-value = 0.724
alternative hypothesis: true location shift is not equal to 0



<font color="#039dfc">

With the p-value of $0.724$, which is higher than the $\alpha$ of $0.05$, we cannot reject the null hypothesis. Therefore we are not able to reject that the mediums are same.


</font>

### 第三大題(共30分)：每小題10分。
### Third Part (Total 30 points): 10 points each

某鄉鎮委託一名衛生所醫師調查當地兩個區域(A, B)地下水的大腸桿菌濃度。兩區域皆隨機選擇7處地下抽水幫浦抽查(每處抽水幫浦在不同時間抽查三次，並取其平均值作為該抽水幫浦的測量結果)，其大腸桿菌濃度(CFU/100mL)測量結果如下表所示。請問不同區域的地下水大腸桿菌濃度是否有顯著差異? (α=0.05) <br>
A town appointed a doctor from health center to investigate the concentrations of Escherichia coli (E. coli) in groundwater of two areas (A, B). They randomly select 7 groundwater pumps for each area (check three times in different periods for each pump, and take average of the three observations as the measurement results). The measurements of the concentrations of E. coli (CFU/100mL) are shown below. Is there significant difference in the concentrations of E. coli between two areas? (α=0.05)

| Area A | Area B |
| ------ | ------ |
| 4.66   | 4.10   |
| 4.31   | 5.30   |
| 4.81   | 5.23   |
| 4.94   | 4.76   |
| 4.12   | 5.29   |
| 5.15   | 5.20   |
| 5.25   | 4.25   |


<font color="#fcba03">

Q8.	請根據提議設定虛無假設與對立假設 <br>
Q8. Please state the null hypothesis and alternative hypothesis according to the question.



</font>

<font color="#039dfc">

### Ans
$H_0$: There is no difference in the medium of concentrations of E. coli between Area A and B <br>
$H_1$: There is a difference in the medium of concentrations of E. coli between Area A and B

</font>

<font color="#fcba03">

Q9.	請判斷此題合適的統計方法為何者？ <br>
Q9. Please decide an appropriate statistical method for this problem.

</font>

<font color="#039dfc">

### Ans
I decide to use the **Wilcoxon Signed-Rank test** for its larger statistical power and to fit the curriculum for this week.

</font>

<font color="#fcba03">

Q10. 計算檢定統計量並做出判斷

</font>

<font color="#039dfc">

### Ans

</font>

In [5]:
area_a <- c(4.66, 4.31, 4.81, 4.94, 4.12, 5.15, 5.25)
area_b <- c(4.10, 5.30, 5.23, 4.76, 5.29, 5.20, 4.25)

wilcox_test3 <- wilcox.test(area_a, area_b, alternative = "two.sided", conf.level = 0.95)

print(wilcox_test3)



	Wilcoxon rank sum exact test

data:  area_a and area_b
W = 19, p-value = 0.535
alternative hypothesis: true location shift is not equal to 0



<font color="#039dfc">

Having a p-value of $0.535$, which is higher than the $\alpha = 0.05$, we failed to reject the null hypothesis, which means that we are not able to neglect that the mediums of two groups are same.

</font>

code provided at https://github.com/lusteven901228/2023_biostatistics/blob/main/HW5/hw5.ipynb